lib/active_record_shards/tasks.rb

```diff
@@ -89,18 +85,20 @@
 
 module ActiveRecordShards
   module Tasks
-    if ActiveRecord::VERSION::MAJOR < 5
-      def self.root_connection(conf)
-        ActiveRecord::Base.send("#{conf['adapter']}_connection", conf.merge('database' => nil))
-      end
-    else
+    if ActiveRecord::VERSION::MAJOR >= 4
       def self.root_connection(conf)
-        # this will trigger rails to load the adapter
         conf = conf.merge('database' => nil)
         resolver = ActiveRecord::ConnectionAdapters::ConnectionSpecification::Resolver.new(ActiveRecord::Base.configurations)
         resolver.spec(conf)
         ActiveRecord::Base.send("#{conf['adapter']}_connection", conf)
       end
+    else
+      def self.root_connection(conf)
+        conf = conf.merge('database' => nil)
+        resolver = ActiveRecordShards::ConnectionSpecification::Resolver.new conf, ActiveRecord::Base.configurations
```

Comment:
can we keep 1 method and do the switch inside of it ?

Reply:
sure, will do